packages/runtime/src/widgets/Form/TextInput.tsx

```diff
@@ -39,12 +41,22 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
     }
   }, [value, formInstance, values?.id, values?.label]);
 
+  const changeHandler = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ): void => setValue(event.target.value);
+
+  // We need to debounce the change handler otherwise state updates will create too many re-renders
+  const debouncedChangeHandler = useMemo(
+    () => debounce(changeHandler, 300),
+    [],
+  );
```

Comment:
@evshi This is good to add debounce in this textInput to avoid frequently unnecessary renders of the component, but at same time, this is very bad experience on typing any values in textInput. If I want to type anything on input, I can't type at my regular speed, I need to press one key, then wait for some time and then I need to press other key, overall to work this, we need to reduce the typing speed. and for end user, this is not an ideal behaviour

Reply:
Yeah this was meant to only debounce the widget state value, not the UI I need to adjust this.

Reply:
Resolved this by moving the debounce to useRegisterBindings with an option to control.